mso/datasource_mso_schema_site_service_graph_node.go

```diff
@@ -0,0 +1,180 @@
+package mso
+
+import (
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/ciscoecosystem/mso-go-client/client"
+	"github.com/ciscoecosystem/mso-go-client/models"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+)
+
+func datasourceMSOSchemaSiteServiceGraphNode() *schema.Resource {
+	return &schema.Resource{
+		Read: datasourceMSOSchemaSiteServiceGraphNodeRead,
+
+		SchemaVersion: version,
+
+		Schema: (map[string]*schema.Schema{
+			"schema_id": &schema.Schema{
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringLenBetween(1, 1000),
+			},
+			"template_name": &schema.Schema{
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringLenBetween(1, 1000),
+			},
+			"service_graph_name": &schema.Schema{
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringLenBetween(1, 1000),
+			},
+			"service_node_name": &schema.Schema{
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringLenBetween(1, 1000),
+			},
+			"service_node_type": &schema.Schema{
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringLenBetween(1, 1000),
+			},
+
+			"site_nodes": &schema.Schema{
+				Type:     schema.TypeList,
+				Optional: true,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"site_id": &schema.Schema{
+							Type:         schema.TypeString,
+							Optional:     true,
+							Computed:     true,
+							ValidateFunc: validation.StringLenBetween(1, 1000),
+						},
+
+						"tenant_name": &schema.Schema{
+							Type:         schema.TypeString,
+							Optional:     true,
+							Computed:     true,
+							ValidateFunc: validation.StringLenBetween(1, 1000),
+						},
+
+						"node_name": &schema.Schema{
+							Type:         schema.TypeString,
+							Optional:     true,
+							Computed:     true,
+							ValidateFunc: validation.StringLenBetween(1, 1000),
+						},
+					},
+				},
+			},
+		}),
+	}
+}
+
+func datasourceMSOSchemaSiteServiceGraphNodeRead(d *schema.ResourceData, m interface{}) error {
+	log.Printf("[DEBUG] Datasource MSO Schema Site Service GraphNode: Beginning Read")
+
+	msoClient := m.(*client.Client)
+
+	schemaId := d.Get("schema_id").(string)
+	cont, err := msoClient.GetViaURL(fmt.Sprintf("api/v1/schemas/%s", schemaId))
+	if err != nil {
+		return err
+	}
+
+	stateTemplate := d.Get("template_name").(string)
```

Comment:
Is it viable to query all the templates first to check if the template is available or if there is more than one template with the same name? or is this unnecessary?

Reply:
Nvm @akinross pointed me to getTemplateServiceGraphCont that already does it. Resolving this..